task6-hello-module-procfs/hello.c

```diff
@@ -0,0 +1,106 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/moduleparam.h>
+#include <linux/printk.h>
+#include <linux/proc_fs.h>
+#include <linux/uaccess.h>
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Kriz");
+MODULE_DESCRIPTION("Hello_world module with simple canculator (sum, sub and mul)");
+MODULE_VERSION("0.1");
+
+#define PROC_BUFFER_SIZE 100
+
+static char   procfs_buffer[PROC_BUFFER_SIZE] = {0};
+static size_t procfs_buffer_size = 0;
+
+static struct proc_dir_entry *proc_file;
+static struct proc_dir_entry *proc_folder;
+
+#define PROC_FILE_NAME "dummy"
+#define PROC_DIR_NAME  "hello"
+
+static ssize_t hello_read(struct file *File, char __user* buffer, size_t count, loff_t *offset)
+{
+	ssize_t to_copy, not_copied, delta;
+	
+	if(0 == procfs_buffer_size)
+	{
+		return 0;
+	}
+	
+	to_copy = min(count, procfs_buffer_size);
+
+	not_copied = copy_to_user(buffer, procfs_buffer, to_copy);
```

Comment:
Per my understanding, if `procfs_buffer_size` variable is maintained and updated in the code below it is assumed that `procfs_buffer` contents can be read by the user space app by multiple reads.
For example 100 bytes buffer can be read by 2 `hello_read` calls with count set to 25 and 75.
But, data is copied to the user provided buffer from the beginning of the `procfs_buffer`, so the same data will be copied twice, what, per my understanding, is not expected.